mmf/utils/inference.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import argparse
+
+import numpy as np
+import omegaconf
+import requests
+import torch
+from mmf.common.registry import registry  # noqa
+from mmf.common.sample import Sample, SampleList
+from mmf.utils.build import build_encoder, build_model, build_processors
+from mmf.utils.checkpoint import load_pretrained_model
+from mmf.utils.general import get_current_device
+from PIL import Image
+
+
+class Inference:
+    def __init__(self, checkpoint_path=None):
```

Comment:
This shouldn't be a script but an API. So what you can do is take the checkpoint_path as an input to init. Let's get rid of the parser.

Reply:
Agreed, script using this API should go to tools.